editor/src/utils/react-performance.ts

```diff
@@ -209,141 +209,188 @@ function failSafeMemoEqualityFunction(componentDisplayName: string, severity: 's
   }
 }
 
+// this function has been adopted from https://github.com/epoberezkin/fast-deep-equal/tree/a33d49ab5cc659e331ff445109f35dd323230d41
```

Comment:
Where in that code does it come from?

Reply:
there's only one source file, https://github.com/epoberezkin/fast-deep-equal/blob/a33d49ab5cc659e331ff445109f35dd323230d41/src/index.jst

should I link that instead?

Reply:
That makes sense, even if it's got some crazy code generation magic behind it. :/